File: Python/directory.py
import os


# Check if a directory exists
def check_dir(dir_name):
    return os.path.exists(dir_name)


# Create a directory
def create_dir(dir_name):
    os.mkdir(dir_name)


# Remove a directory
def remove_dir(dir_name):
    os.rmdir(dir_name)


# Change the current working directory
def change_dir(dir_name):
    os.chdir(dir_name)


# Get all the files and folders in a directory
def get_all_folder_files(dir_name):
    return os.listdir(dir_name)


# Get all the folders only in a directory
def get_folders_only(dir_name):
    return [f for f in os.listdir(dir_name) if os.path.isdir(os.path.join(dir_name, f))]


# Get all the files only in a directory
def get_files_only(dir_name):
    return [f for f in os.listdir(dir_name) if os.path.isfile(os.path.join(dir_name, f))]


# Get the current working directory
def get_current_directory():
    return os.getcwd()


def main():
    # Check if the directory exists
    print(check_dir("/"))
    # Create a directory
    create_dir("source/")
    # Get all the files and folders in a directory
    print(get_all_folder_files("source/"))
    # Get all the folders only in a directory
    print(get_folders_only("source/"))
    # Get all the files only in a directory
    print(get_files_only("source/"))
    # Get the current working directory
    print(get_current_directory())
    # Remove a directory
    remove_dir("source/")
    # Change the current working directory
    change_dir("/")

main()

File: Rust/src/directory.rs
#![allow(clippy::needless_return)]
use std::env;

fn main() {
    // Get the current working directory
    println!("{}", get_cwd());
    // Create a new directory
    create_directory("new_dir/");
    // Change the current working directory
    change_cwd("new_dir/");
    // Get all the folders in the current working directory
    println!("{:?}", get_folders_in_current_path());
    // Get all the folders in the given directory
    println!("{:?}", get_folders_from_path("new_dir/"));
    // Delete a directory
    remove_directory("new_dir/");
}

// Get the current working directory and return it as a String
fn get_cwd() -> String {
    return env::current_dir().unwrap().display().to_string();
}

// Change the current working directory to the given path
fn change_cwd(path: &str) {
    env::set_current_dir(path).unwrap();
}

// Get all the folders only in the current working directory
fn get_folders_in_current_path() -> Vec<String> {
    let mut folders = Vec::new();
    for entry in env::current_dir().unwrap().read_dir().unwrap() {
        let entry = entry.unwrap();
        if entry.file_type().unwrap().is_dir() {
            folders.push(entry.path().display().to_string());
        }
    }
    return folders;
}

// Get all the folders only from the given path
fn get_folders_from_path(path: &str) -> Vec<String> {
    let mut folders = Vec::new();
    for entry in std::fs::read_dir(path).unwrap() {
        let entry = entry.unwrap();
        if entry.file_type().unwrap().is_dir() {
            folders.push(entry.path().display().to_string());
        }
    }
    return folders;
}

// Create a directory
fn create_directory(path: &str) {
    std::fs::create_dir(path).unwrap();
}

// Remove a directory
fn remove_directory(path: &str) {
    std::fs::remove_dir(path).unwrap();
}
